fe/fe-core/src/main/java/org/apache/doris/transaction/PublishVersionDaemon.java

```diff
@@ -190,16 +190,18 @@ private void publishVersion() throws UserException {
                             LOG.warn("Database [{}] has been dropped.", transactionState.getDbId());
                             continue;
                         }
-                        db.readLock();
-                        try {
-                            for (int i = 0; i < transactionState.getTableIdList().size(); i++) {
-                                long tableId = transactionState.getTableIdList().get(i);
-                                Table table = db.getTable(tableId);
-                                if (table == null || table.getType() != Table.TableType.OLAP) {
-                                    LOG.warn("Table [{}] in database [{}] has been dropped.", tableId, db.getFullName());
-                                    continue;
-                                }
-                                OlapTable olapTable = (OlapTable) table;
+
+
+                        for (int i = 0; i < transactionState.getTableIdList().size(); i++) {
+                            long tableId = transactionState.getTableIdList().get(i);
+                            Table table = db.getTable(tableId);
+                            if (table == null || table.getType() != Table.TableType.OLAP) {
+                                LOG.warn("Table [{}] in database [{}] has been dropped.", tableId, db.getFullName());
+                                continue;
+                            }
+                            OlapTable olapTable = (OlapTable) table;
+                            olapTable.readLock();
```

Comment:
Is it more safe to lock all tables at once outside the for loop?

Reply:
I think the final result is same if the table is dropped before loop or in loop, we just make sure that it is thread safe, and we has already add committed txn check if user want to drop table during loading data